src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -31,26 +32,49 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
 
         protected virtual async Task<bool> HandleRemoteCallbackAsync()
         {
-            var authResult = await HandleRemoteAuthenticateAsync();
-            if (authResult != null && authResult.Skipped)
+            AuthenticateResult authResult = null;
+            Exception exception = null;
+
+            try
             {
-                return false;
+                authResult = await HandleRemoteAuthenticateAsync();
+                if (authResult != null && authResult.Skipped == true)
+                {
+                    return false;
+                }
+                else if (authResult == null)
+                {
+                    exception = new InvalidOperationException("Invalide return state, unable to redirect.");
+                }
+                else if (!authResult.Succeeded)
+                {
+                    exception = authResult?.Failure ??
+                                new InvalidOperationException("Invalide return state, unable to redirect.");
+                }
+            }
+            catch (Exception ex)
+            {
+                exception = ex;
             }
-            if (authResult == null || !authResult.Succeeded)
+
+            if (exception != null)
             {
-                var errorContext = new FailureContext(Context, authResult?.Failure ?? new Exception("Invalid return state, unable to redirect."));
-                Logger.RemoteAuthenticationError(errorContext.Failure.Message);
+                Logger.RemoteAuthenticationError(exception.Message);
+                var errorContext = new FailureContext(Context, exception);
                 await Options.Events.RemoteFailure(errorContext);
+
                 if (errorContext.HandledResponse)
                 {
                     return true;
                 }
-                if (errorContext.Skipped)
+                else if (errorContext.Skipped)
                 {
                     return false;
                 }
-
-                throw new AggregateException("Unhandled remote failure.", errorContext.Failure);
+                else
```

Comment:
Same thing, easier to read if the bottom is a clear throw, without having to scroll up to figure out what the else is for